lib/common.sh

```diff
@@ -849,6 +798,52 @@ common::set_cloud_binaries () {
   fi
 }
 
+###############################################################################
+# sendmail/mailer front end.
+# @optparam html - Send html formatted
+# @param to - To
+# @param from - From
+# @param reply_to - Reply To
+# @param subject - Subject
+# @param cc - cc
+# @param file - file to send
+#
+common::sendmail () {
+  local cc_arg
+  local html=0
+
+  while [[ "$#" -gt 0 ]]; do
+    case "$1" in
+      -h) html=1; shift ;;
+       *) break ;;
+    esac
+  done
+
+  local to="$1"
+  local from="$2"
+  local reply_to="$3"
+  local subject="$4"
+  local cc="$5"
+  local file="$6"
+
+  # Conditional execute
+  if type -t security_layer::gmailer >/dev/null 2>&1; then
```

Comment:
rather than doing this, maybe the security layer should just override `common::sendmail` entirely?

similarly, we could have no-op versions of `hosted_map` and `auth_check` which we just override - then we don't need to do this funny dance of checking whether the functions exist.